src/hooks/Guilds/contracts/useDecodedCall.test.ts

```diff
@@ -0,0 +1,68 @@
+import { renderHook } from '@testing-library/react-hooks';
+import { BigNumber } from 'ethers';
+import { SupportedAction } from 'old-components/Guilds/ActionsBuilder/types';
+import { ZERO_ADDRESS, ZERO_HASH } from 'utils';
+import { useDecodedCall } from './useDecodedCall';
+
+const mockChainId = 1337;
+const contractAddress = '0x0000000000000000000000000000000000000001';
+const erc20TransferData =
+  '0xa9059cbb000000000000000000000000f960cec172d9ea3c0233d1caaceace4b597cbccf0000000000000000000000000000000000000000000000000000000077359400';
+
+jest.mock('@web3-react/core', () => ({
+  useWeb3React: () => ({
+    chainId: mockChainId,
+  }),
+}));
+
+jest.mock('./useRichContractData', () => ({
+  useRichContractData: () => ({
+    contracts: [],
+  }),
+}));

```

Comment:
We should add a test for decoding via rich contract data also but we can do this later

Reply:
Yeah, we need to add more tests to test all the individual functions (`decodeCallUsingEthersInterface`, `getContractInterfaceFromRichContractData`, `getContractFromKnownSighashes` etc.). I didn't do it in this PR as it's a bit complicated to build the input data and mock everything. We should tackle that soon though.